### PR TITLE
Update Spring Boot Quickstarts to latest version

### LINKED ISSUE #716: Update Spring Boot Quickstarts to latest version
Our Spring Boot Quickstarts are based upon spring boot versions that were released a year ago. We need to update them to the latest version of spring boot.

1. Update Spring Boot version in Quickstart to latest version
2. Check if Quickstart is building fine after version bump
3. Test if upgraded quickstart is working okay with jkube. Check `mvn k8s:build`, `mvn k8s:resource`, `mvn k8s:apply` goals with a local Kubernetes cluster(minikube/kind)

https://github.com/eclipse/jkube/tree/master/quickstarts/maven/spring-boot
https://github.com/eclipse/jkube/tree/master/quickstarts/maven/spring-boot-crd
https://github.com/eclipse/jkube/tree/master/quickstarts/maven/spring-boot-camel-complete
https://github.com/eclipse/jkube/tree/master/quickstarts/maven/spring-boot-dekorate
https://github.com/eclipse/jkube/tree/master/quickstarts/maven/spring-boot-with-jib
https://github.com/eclipse/jkube/tree/master/quickstarts/maven/spring-boot-watch

### PR DESCRIPTION
**Pull request by theexplorist**
## Description
closes #716 

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [x] I tested my code in Kubernetes
 - [x] I tested my code in OpenShift

### CODE REVIEW

**Comment by theexplorist** [comment]
spring-boot
dekorate
watch
jib

**Comment by manusa** [comment]
> The issues are in crd and camel quickstarts, the containers are up but I think that is end point issue.

The camel endpoint should be visible when running `curl $(kubectl get routes.route.openshift.io spring-boot-camel-complete -o jsonpath='{.spec.host}')"/camel/hello-camel"`, did you check this path?

You also need to check the logs for the order generation and processing messages.

File: CHANGELOG.md [hunk 1]
```diff
@@ -42,6 +42,7 @@ Usage:
 * Fix #756: Service re-apply error happening during `k8s:watch`
 * Fix #758: QuarkusHealthCheckEnricher: default health path value is outdated
 * Fix #777: Error processing tar file(archive/tar: missed writing 4096 bytes) on Docker 1.12.1
+* Fix #716: Update Spring Boot Quickstarts to latest version
```
Comment (manusa) [inline CHANGELOG.md:45]: This needs to go below the `1.5.0-SNAPSHOT header`

**Comment by manusa** [comment]
> The issues are in crd and camel quickstarts, the containers are up but I think that is end point issue.

You won't be able to test the CRD version on Red Hat's OpenShift Dev sandbox since (IIRC) you can't deploy CRDs there. You can check that one on Minikube

**Review by rohanKanojia — APPROVED** [review]

**Comment by theexplorist** [comment]
# Kuberenetes Cluster Testing

## Camel-Complete
Camel

## Spring-Boot
Spring-boot deploy
Spring-boot cont

## CRD
CRD - 1
CRD - Minikube 
CRD command

## JIB
JIB

**Review by manusa — APPROVED** [review]

**Pull request merged**